app/Http/Controllers/Admin/CategoryController.php

```diff
@@ -19,27 +21,29 @@ public function __construct()
 
     public function index()
     {
-        $data['categories'] = $this->objCategory->getAllCategories();
+        $data['categories'] = $this->objCategory->getAllCategoriesInfo();
         return view('pages.admins.categories.index', $data);
     }
 
-    public function store(Request $request)
+    public function store(CategoryRequest $request)
     {
-        $request->validate([
-            'category_name' => 'required',
-            'category_active' => 'required',
-        ]);
-        $this->objCategory->create($request->all());
-        return redirect()->route('categories');
+        $response = $this->objCategory->create($request->all());
+        return redirect()->route('categories')->with($response['type'],$response['message']);
```

Comment:
please put a space after the comma 

return redirect()->route('categories')->with($response['type'], $response['message']);